app/models/reducers/external_reducer.rb

```diff
@@ -22,10 +22,14 @@ class ExternalReducerFailed < StandardError; end
     end
 
     def reduce_into(extractions, reduction)
-      #TODO: is this the right idea here?
-      #TODO: this has to be a store thing too
       if url
-        response = RestClient.post(url, extractions.to_json, {content_type: :json, accept: :json})
+        response = if default_reduction?
+          RestClient.post(url, extractions.to_json, {content_type: :json, accept: :json})
+        elsif running_reduction?
+          RestClient.post(url, { extracts: extractions, store: reduction.store }.to_json, {content_type: :json, accept: :json})
+        else
+          nil
```

Comment:
If this returns nil then it will crash a few lines later with NoMethod code for nil:NilClass.

Reply:
Technically it shouldn't be possible to hit that `else` at all (because the only two modes are `default_reduction` and `running_reduction`) so the `else` is only really there for completeness. Would it be better to remove it?

Reply:
Yes, or raise some other "this is not expected to every happen" error to clarify that